src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -105,6 +108,20 @@ async def _publisher(self) -> None:
                     )
             self._events_to_send.task_done()
 
+    async def _send_terminate_messages_to_dispatchers(self) -> None:
+        event = TERMINATE_MSG
+        await asyncio.gather(
+            *(
+                self._router_socket.send_multipart([identity, b"", event])
+                for identity in self._dispatchers_connected
+            )
+        )
+
+    async def _terminate_all_dispatchers(self) -> None:
+        await self._send_terminate_messages_to_dispatchers()
+        await asyncio.sleep(self.WAIT_PERIOD_FOR_GRACEFUL_SHUTDOWN)
+        await self._ensemble.cancel()
```

Comment:
Is it possible to log here which realizations that needs an extra kill?

Reply:
If all dispatchers happily terminate before the grace period, we should not have to wait for the grace period.